Jenkinsfile

```diff
@@ -512,6 +513,10 @@ pipeline {
   }
   post {
     cleanup {
+            withAzureCredentials(path: "${homePath}", credentialsFile: '.credentials.json') {
+                powershell label: "Checking and removing any Azure related resource groups",
+                    script: "az group list --query \"[?name | starts_with(@,'${AZURE_RESOURCE_GROUP_PREFIX})']\" --out json|jq .[].name | ForEach {az group delete --name $_}"
```

Comment:
Before `az group list` can run, 

1. [the Service Principal credentials in `.credentials.json` will need to be used to log in](https://docs.microsoft.com/en-us/cli/azure/authenticate-azure-cli#sign-in-with-a-service-principal). Something like the following (this needs checking/testing to ensure sensitive details are not captured in logs or displayed in CI)

```sh
az login --service-principal --username <clientId> --password <clientSecret> --tenant <tenantId>
```

2. Since a tenant can contain multiple subscriptions, the active subscription to work with should be set as the one from `.credentials.json`

```sh
az account set --subscription <subscriptionId>
```

Now, `az group list` is good to go

Reply:
```
jq .[].name
```
Could avoid needing to install `jq`, if it's just for this, using PowerShell's [`ConvertFrom-Json`](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/convertfrom-json?view=powershell-7.1). Also would want [`--nowait` and `--yes` when deleting groups](https://docs.microsoft.com/en-us/cli/azure/group?view=azure-cli-latest#az_group_delete). Something like

```powershell
.. | ConvertFrom-Json | ForEach { az group delete --name $_.name --no-wait --yes  }
```

Reply:
Would need a bash/sh variant too, for Linux build hosts?